onnxruntime/core/providers/cpu/tensor/slice.h

```diff
@@ -11,27 +11,18 @@ class SliceBase {
  protected:
   SliceBase(const OpKernelInfo& info) {
     has_axes_ = info.GetAttrs("axes", axes_).IsOK();
-
-    ORT_ENFORCE(info.GetAttrs("starts", starts_).IsOK(), "Invalid 'starts' attribute value");
-    ORT_ENFORCE(info.GetAttrs("ends", ends_).IsOK(), "Invalid 'ends' attribute value");
-
-    if (has_axes_) {
-      if (axes_.size() > starts_.size())
-        ORT_THROW("'axes' has more entries than the 'starts' attribute holds");
-      if (axes_.size() > ends_.size())
-        ORT_THROW("'axes' has more entries than the 'ends' attribute holds");
-    }
+    info.GetAttrs("starts", starts_);
+    info.GetAttrs("ends", ends_);
   }
 
   Status PrepareForCompute(const size_t dimension_count, const std::vector<int64_t>& input_dimensions,
                            std::vector<int64_t>& starts, std::vector<int64_t>& output_dims) const;
 
-  std::vector<int64_t> axes_;
-  bool has_axes_;
-  std::vector<int64_t> starts_, ends_;
+  mutable bool has_axes_;
+  mutable std::vector<int64_t> starts_, ends_, axes_;
```

Comment:
Please remove the mutable keyword. The Compute function must be thread safe

Reply:
Thanks for the comment!
mutable here is to allow vectors to be modified in compute function, otherwise compile will fail.
As of thread-safety, the current implementation of computes all operate on an OpKernelContext instance with memory allocation and writing, I am not seeing any code working on assuring thread-safety. So how they achieved thread-safety anyway?

Reply:
By default, in most cases, the kernel should be stateless.